File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/INTRODUCCION/Ejercicio7Py_Java/Mascota.java
package Ejercicio7Py_Java;

public class Mascota {
    private String nombre;
    private String tipo;
    private int energia;

    public Mascota(String nombre, String tipo, int energia) {
        this.nombre = nombre;
        this.tipo = tipo;
        this.energia = energia;
    }

    public void alimentar() {
        int energiaAntes = energia;
        energia = Math.min(energia + 20, 100);
        System.out.println(nombre + " fue alimentado. Energía: " + energiaAntes + "," + energia);
    }

    public void jugar() {
        int energiaAntes = energia;
        energia = Math.max(energia - 15, 0);
        System.out.println(nombre + " jugó. Energía: " + energiaAntes + " , " + energia);
    }

    public void mostrarEnergia() {
        System.out.println("Energía de " + nombre + ": " + energia);
    }
    public void mostrarDatos() {
    System.out.println("Nombre: " + nombre + ", Tipo: " + tipo + ", Energía: " + energia);
}

}

File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/INTRODUCCION/Ejercicio7Py_Java/Mascota.py
class Mascota:
    def __init__(self, nombre, tipo, energia):
        self.nombre = nombre
        self.tipo = tipo
        self.energia = energia

    def alimentar(self):
        energia_antes = self.energia
        self.energia = min(self.energia + 20, 100)
        print(f"{self.nombre} fue alimentado. Energía: {energia_antes} , {self.energia}")

    def jugar(self):
        energia_antes = self.energia
        self.energia = max(self.energia - 15, 0)
        print(f"{self.nombre} jugó. Energía: {energia_antes} , {self.energia}")

    def mostrarEnergia(self):
        print(f"Energía de {self.nombre}: {self.energia}")
